soc/arm/st_stm32/stm32l4/power.c

```diff
@@ -20,89 +20,140 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
 
+
+
 /* select MSI as wake-up system clock if configured, HSI otherwise */
 #if STM32_SYSCLK_SRC_MSI
 #define RCC_STOP_WAKEUPCLOCK_SELECTED LL_RCC_STOP_WAKEUPCLOCK_MSI
 #else
 #define RCC_STOP_WAKEUPCLOCK_SELECTED LL_RCC_STOP_WAKEUPCLOCK_HSI
 #endif
 
-/* Invoke Low Power/System Off specific Tasks */
-__weak void pm_state_set(enum pm_state state, uint8_t substate_id)
+
+
+void enter_low_power_mode(void)
 {
-	if (state != PM_STATE_SUSPEND_TO_IDLE) {
-		LOG_DBG("Unsupported power state %u", state);
-		return;
+
+	/* Configure CPU core */
+	/* Enable CPU deep sleep mode */
+#ifndef CONFIG_DEBUG
+	LL_LPM_EnableDeepSleep();
```

Comment:
Is this required before shutdown/standby ?

Reply:
yes